tests/array.rs

```diff
@@ -807,6 +816,49 @@ fn owned_array_with_stride() {
     assert_eq!(a.strides(), &[1, 4, 2]);
 }
 
+#[test]
+fn owned_array_discontiguous() {
+    use ::std::iter::repeat;
+    let v: Vec<_> = (0..12).flat_map(|x| repeat(x).take(2)).collect();
+    let dim = (3, 2, 2);
+    let strides = (8, 4, 2);
+
+    let a = Array::from_shape_vec(dim.strides(strides), v).unwrap();
+    assert_eq!(a.strides(), &[8, 4, 2]);
+    println!("{:?}", a.iter().cloned().collect::<Vec<_>>());
+    itertools::assert_equal(a.iter().cloned(), 0..12);
+}
+
+#[test]
+fn owned_array_discontiguous_drop() {
+    use ::std::rc::Rc;
+    use ::std::cell::RefCell;
+    use ::std::collections::BTreeSet;
+
+    struct InsertOnDrop<T: Ord>(Rc<RefCell<BTreeSet<T>>>, Option<T>);
+    impl<T: Ord> InsertOnDrop<T> {
+        fn new(set: Rc<RefCell<BTreeSet<T>>>, value: T) -> Self {
+            InsertOnDrop(set, Some(value))
+        }
+    }
+    impl<T: Ord> Drop for InsertOnDrop<T> {
+        fn drop(&mut self) {
+            let InsertOnDrop(ref set, ref mut value) = *self;
+            set.borrow_mut().insert(value.take().expect("double drop!"));
+        }
+    }
+
+    let set = Rc::new(RefCell::new(BTreeSet::new()));
+    {
+        let v: Vec<_> = (0..12).map(|x| InsertOnDrop::new(set.clone(), x)).collect();
+        let mut a = Array::from_shape_vec((2, 6), v).unwrap();
+        // discontiguous and non-zero offset
+        a.slice_inplace(s![.., 1..]);
+    }
+    // each item was dropped
```

Comment:
Will you please change this comment to say "each item was dropped exactly once"?